awserr_test.go

```diff
@@ -0,0 +1,340 @@
+package awsbase
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws/awserr"
+)
+
+func TestIsAwsErr(t *testing.T) {
+	testCases := []struct {
+		Name     string
+		Err      error
+		Code     string
+		Message  string
+		Expected bool
```

Comment:
I don't see Expected set on most of the test cases, are you relying on a default behavior?

Reply:
In Go, each type has an associated zero value (https://tour.golang.org/basics/12) and for `bool` it is `false`. I'm not sure I necessarily have good reference of linters that check for this specifically or documentation for whether or not its idiomatic to rely on zero values in composite literals other than maybe [this section of Effective Go](https://golang.org/doc/effective_go.html#composite_literals). If we'd prefer explicitly writing out `Expected: false,` in places though, can certainly update this.

Reply:
It's consistent so I think this is ok. I have a general preference for explicit defaults -- but I'm still getting used to the Go testing framework, so I'm not going to be pushy about what that should look like.